spec/features/saml/loa3_sso_spec.rb

```diff
@@ -118,15 +118,15 @@
 
       it 'prompts for OTP at sign in' do
         saml_authn_request = auth_request.create(loa3_with_bundle_saml_settings)
-
         visit saml_authn_request
-
         sign_in_live_with_2fa(user)
+        loa3_sp_session(request_url: saml_authn_request)
```

Comment:
Since this test is going through SamlIdpController, it should not be necessary to stub the session. In fact, it's discouraged because it might be masking a bug. We want to simulate the user experience as closely as possible.